angr/analyses/reaching_definitions/dep_graph.py

```diff
@@ -145,12 +145,21 @@ def add_dependencies_for_concrete_pointers_of(self,
             )
         ))
 
+        #concretize ddresses where possible
+        concrete_known_pred_addresses = []
+        for address in known_predecessor_addresses:
+            if isinstance(address, claripy.ast.Base):
+                if address.concrete:
+                    concrete_known_pred_addresses.append(address._model_concrete.value)
+            else:
```

Comment:
What's the type of `address` under `else`?

Reply:
`int` type addresses

Reply:
This line -> https://github.com/angr/angr/blob/79f484e78aca77bcc00f79374dbb7a74ac0bdcc9/angr/analyses/reaching_definitions/dep_graph.py#L161
would throw an error if there were any `claripy.ast.Base` objects in the list.
The above loop just makes sure that everything is safely comparable.